src/main/java/com/example/provider/json/SerializedTrip.java

```diff
@@ -50,6 +56,10 @@ abstract static class Builder {
 
     abstract Builder setVehicleId(String vehicleId);
 
+    abstract Builder setRouteList(List<LatLng> routes);
+
+    abstract Builder setCurrentRouteSegmentTraffic(ConsumableTrafficPolyline routeTraffic);
```

Comment:
Are we using the current route segment traffic for anything? The segment is encoded and is used to represent just the traffic along the route which I don't think we are using when drawing the polyline so can we remove it?